source/postprocess/geoid.cc

```diff
@@ -201,52 +195,46 @@ namespace aspect
                                       update_quadrature_points |
                                       update_JxW_values);
 
-      std::vector<double> topo_values( quadrature_formula_face.size());
-
-      // vectors to store the location, infinitesimal area, and dynamic topography associated with each quadrature point of each surface and bottom cell respectively.
+      // Vectors to store the location, infinitesimal area, and topography associated with each quadrature point of each surface and bottom cell respectively.
       std::vector<std::pair<Point<3>,std::pair<double,double> > > surface_stored_values;
       std::vector<std::pair<Point<3>,std::pair<double,double> > > CMB_stored_values;
 
-      // loop over all of the boundary cells and if one is at
-      // surface or CMB, evaluate the dynamic topography vector there.
+      // Loop over all of the boundary cells and if one is at
+      // surface or CMB, evaluate the topography vector there.
       for (const auto &cell : this->get_dof_handler().active_cell_iterators())
         if (cell->is_locally_owned() && cell->at_boundary())
           {
-            // see if the cell is at the *top* boundary or CMB, not just any boundary
             unsigned int face_idx = numbers::invalid_unsigned_int;
             bool at_upper_surface = false;
             {
               for (unsigned int f=0; f<GeometryInfo<3>::faces_per_cell; ++f)
                 {
                   if (cell->at_boundary(f) && this->get_geometry_model().depth (cell->face(f)->center()) < cell->face(f)->minimum_vertex_distance()/3)
```

Comment:
I just realized that this condition will no longer work once you have a real free surface. Currently the postprocessor checks if the midpoint of the current cell face is close to a depth of 0 (for top) or the maximal_depth (CMB) to check if it is the top or bottom surface. This condition will need to be replaced by the following (which is better anyway):

```suggestion
                  if (cell->at_boundary(f) && cell->face(f)->boundary_id() == this->get_geometry_model().translate_symbolic_boundary_name_to_id("top"))
```

or the following for the CMB check: 


```
                  if (cell->at_boundary(f) && cell->face(f)->boundary_id() == this->get_geometry_model().translate_symbolic_boundary_name_to_id("bottom"))
```

Please have a look through the postprocessor and replace the old conditions with the new ones (I think there are 4 places that need to be replaced). Otherwise you may see wrong results from the postprocessor with a free surface.

Reply:
Thanks for your quick reviewing!